analysis/blacklist/blacklist.go

```diff
@@ -0,0 +1,223 @@
+package blacklist
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/ocmdev/mgosec"
+	bl "github.com/ocmdev/rita-bl"
+	blDB "github.com/ocmdev/rita-bl/database"
+	"github.com/ocmdev/rita-bl/list"
+	"github.com/ocmdev/rita-bl/sources/lists"
+	"github.com/ocmdev/rita-bl/sources/rpc"
+	"github.com/ocmdev/rita/config"
+	"github.com/ocmdev/rita/database"
+	log "github.com/sirupsen/logrus"
+	mgo "gopkg.in/mgo.v2"
+)
+
+type resultsChan chan map[string][]blDB.BlacklistResult
+
+//BuildBlacklistedCollections builds the blacklisted sources,
+//blacklisted destinations, blacklist hostnames, and blacklisted urls
+//collections
+func BuildBlacklistedCollections(res *database.Resources) {
+	//capture the current value for the error closure below
+	currentDB := res.DB.GetSelectedDB()
+
+	blDB, err := blDB.NewMongoDB(res.System.DatabaseHost, mgosec.None, "rita-bl")
+	if err != nil {
+		res.Log.Error(err)
+		fmt.Println("\t[!] Could not connect to blacklist database")
+		return
+	}
+
+	//set up rita-blacklist
+	ritaBL := bl.NewBlacklist(
+		blDB,
+		func(err error) { //error handler
+			res.Log.WithFields(log.Fields{
+				"db": currentDB,
+			}).Error(err)
+		},
+	)
+
+	//set up the lists to check against
+	ritaBL.SetLists(buildBlacklists(res.System)...)
+
+	//set up remote calls to check against
+	ritaBL.SetRPCs(buildBlacklistRPCS(res)...)
+
+	//update the lists
+	ritaBL.Update()
+
+	//get our data sources
+	ssn := res.DB.Session.Copy()
+	defer ssn.Close()
+
+	uniqueSourcesAggregation := getUniqueIPFromUconnPipeline("src")
+	uniqueDestAggregation := getUniqueIPFromUconnPipeline("dst")
+
+	uniqueSourceIter := res.DB.AggregateCollection(
+		res.System.StructureConfig.UniqueConnTable,
+		ssn,
+		uniqueSourcesAggregation,
+	)
+	uniqueDestIter := res.DB.AggregateCollection(
+		res.System.StructureConfig.UniqueConnTable,
+		ssn,
+		uniqueDestAggregation,
+	)
+	hostnamesIter := ssn.DB(currentDB).C(res.System.DNSConfig.HostnamesTable).
+		Find(nil).Iter()
+	urlIter := ssn.DB(currentDB).C(res.System.UrlsConfig.UrlsTable).
+		Find(nil).Iter()
+
+	//create the collections
+	sourceIPs := res.System.BlacklistedConfig.SourceIPsTable
+	destIPs := res.System.BlacklistedConfig.DestIPsTable
+	hostnames := res.System.BlacklistedConfig.HostnamesTable
+	urls := res.System.BlacklistedConfig.UrlsTable
+
+	collections := []string{sourceIPs, destIPs, hostnames, urls}
+	for _, collection := range collections {
+		ssn.DB(currentDB).C(collection).Create(&mgo.CollectionInfo{
+			DisableIdIndex: true,
+		})
+	}
+
+	//create the data
+	//TODO: refactor these into modules
+	bufferSize := 1000
```

Comment:
This can be a config option, but I don't see any reason why it should be changed. It is the amount of entries to check against rita-bl at a time. The chunking is mainly for rita-bl RPC's such as google safebrowsing. Some RPC's can take advantage of this chunked behavior when querying the internet. The proper thing to do here is probably change it to a constant in the blacklist package.